shotover-proxy/src/transforms/cassandra/sink_cluster/mod.rs

```diff
@@ -229,6 +232,18 @@ fn create_query(messages: &Messages, query: &str, version: Version) -> Result<Me
 
 impl CassandraSinkCluster {
     async fn send_message(&mut self, mut messages: Messages) -> ChainResponse {
+        if let Some(message) = messages.first() {
+            if let Ok(Metadata::Cassandra(CassandraMetadata { version, .. })) = message.metadata() {
+                self.version = version;
+            } else {
+                return Err(anyhow!(
+                    "Failed to extract cassandra version from incoming message: Not a valid cassandra message"
+                ));
+            }
+        } else {
+            return Ok(vec![]);
```

Comment:
Is this an early return if there is no messages?

Reply:
yeah

Reply:
Wouldn't we want to run the checks for updating keyspaces and node list regardless? Though I can't think of a situation where a transform chain would run without any messages, seems reasonable to run the update stuff anyway.

Reply:
Ah very true, good catch. Maybe make a note of it?